### PR TITLE
Fix nick autocomplete

### LINKED ISSUE #1119: JS exception when typing for a private message
To reproduce:

- Open your browser JS console
- Open a private message window (`/whois` or click, didn't try other methods)
- Start typing and enjoy the following in your console:
  screen shot 2017-04-30 at 04 23 16

### PR DESCRIPTION
**Pull request by xPaw**
Fixes #1119.

Also simplifies code by getting rid of the `move` function.

### CODE REVIEW

**Comment by MaxLeiter** [comment]
Can confirm works on top of master 👍

File: client/js/lounge.js [hunk 1]
```diff
@@ -1534,17 +1535,6 @@ $(function() {
...
-		if (new_index >= array.length) {
-			var k = new_index - array.length;
-			while ((k--) + 1) {
-				this.push(undefined);
-			}
-		}
-		array.splice(new_index, 0, array.splice(old_index, 1)[0]);
-		return array;
-	}
-
```
Comment (astorije) [inline client/js/lounge.js]: Shouldn't we get rid of https://github.com/thelounge/lounge/blob/master/client/js/lounge.js#L303-L305 as well then? Also, why was that a `typeof` check on a function of ours?!

Comment (xPaw) [inline client/js/lounge.js]: That code was replaced, look at this diff line 303. `typeof` because bad Shout code.

Comment (astorije) [inline client/js/lounge.js]: Next step, I learn to read.

**Pull request merged**